.pre-commit-config.yaml

```diff
@@ -0,0 +1,16 @@
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v2.3.0
+  hooks:
+  - id: end-of-file-fixer
+  - id: trailing-whitespace
+- repo: https://github.com/psf/black
+  rev: 22.10.0
+  hooks:
+  - id: black
+    args: ['--check', '--diff', '.']
```

Comment:
I'd remove these args. The make this hook substantially less useful. By default, this hook formats the code with black if necessary. With these args, the hook just warns that black needs to be run manually.

Reply:
ftr, I do a lot of `git add -p` style commits, so running `black` on commit means my code gets moved around while I still have the editor open - depending on what other work is going on that has the potential for losing work. Hence the specific choice of only doing `--check` here (and `--diff` to see what needs changing).

Reply:
I always use `git add -p` and I've never seen any data loss due to using `pre-commit`. Usually it'll stash your changes before running black, and if your stash and its changes have conflicts, it just keeps your version around.